catroid/src/test/java/org/catrobat/catroid/test/content/actions/ChangeVariableActionTest.kt

```diff
@@ -0,0 +1,78 @@
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.test.content.actions
+
+import org.junit.Assert
+import org.catrobat.catroid.ProjectManager
+import org.catrobat.catroid.content.Project
+import org.catrobat.catroid.content.SingleSprite
+import org.catrobat.catroid.content.Sprite
+import org.catrobat.catroid.formulaeditor.Formula
+import org.catrobat.catroid.formulaeditor.UserVariable
+import org.catrobat.catroid.test.MockUtil
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+import java.util.concurrent.ThreadLocalRandom
+
+@RunWith(Parameterized::class)
+class ChangeVariableActionTest(private val formula: Formula?, private val variable: UserVariable?, private val expectedValue: Double) {
+    private lateinit var testSprite: Sprite
+    private lateinit var userVariable: UserVariable
+    private lateinit var project: Project
+
+    companion object {
+        @JvmStatic
+        @Parameterized.Parameters
+        fun parameters() = listOf(
+            arrayOf(Formula(CHANGE_VARIABLE_VALUE), UserVariable(TEST_USERVARIABLE, INITIALIZED_VALUE), EXPECTED_NEW_VALUE),
+            arrayOf(Formula(CHANGE_VARIABLE_VALUE), null, INITIALIZED_VALUE),
+            arrayOf(Formula(CHANGE_VARIABLE_VALUE.toString()), UserVariable(TEST_USERVARIABLE, INITIALIZED_VALUE), EXPECTED_NEW_VALUE),
+            arrayOf(Formula(NOT_NUMERICAL_STRING), UserVariable(TEST_USERVARIABLE, INITIALIZED_VALUE), INITIALIZED_VALUE),
+            arrayOf(null, UserVariable(TEST_USERVARIABLE, INITIALIZED_VALUE), INITIALIZED_VALUE),
+            arrayOf(Formula(Double.NaN), UserVariable(TEST_USERVARIABLE, INITIALIZED_VALUE), INITIALIZED_VALUE)
+        )
+
+        private const val NOT_NUMERICAL_STRING = "changeVariable"
+        private const val TEST_USERVARIABLE = "testUservariable"
+        private val CHANGE_VARIABLE_VALUE = ThreadLocalRandom.current().nextDouble(-100.0, 100.0)
+        private val INITIALIZED_VALUE = ThreadLocalRandom.current().nextDouble(-100.0, 100.0)
+        private val EXPECTED_NEW_VALUE = INITIALIZED_VALUE + CHANGE_VARIABLE_VALUE
+    }
+
+    @Before
+    fun setUp() {
+        testSprite = SingleSprite("testSprite")
+        project = Project(MockUtil.mockContextForProject(), "testProject")
+        ProjectManager.getInstance().currentProject = project
+        userVariable = variable ?: UserVariable(TEST_USERVARIABLE, INITIALIZED_VALUE)
```

Comment:
The "userVariable" is always UserVariable(TEST_USERVARIABLE, INITIALIZED_VALUE) except test-case 1.
As discussed, please edit the test-case and delete this variable.